packages/webapp/pages/posts/[id].tsx

```diff
@@ -50,30 +45,27 @@ export interface Props {
   postData?: PostData;
 }
 
+const CONTENT_MAP: Record<PostType, typeof PostContent> = {
+  article: PostContent,
+  share: SquadPostContent,
+};
+
 interface PostParams extends ParsedUrlQuery {
   id: string;
 }
 
 const PostPage = ({ id, postData }: Props): ReactElement => {
   const [position, setPosition] =
     useState<CSSProperties['position']>('relative');
-  const { tokenRefreshed } = useContext(AuthContext);
   const router = useRouter();
   const { isFallback } = router;
 
-  if (!isFallback && !id) {
-    return <Custom404 />;
-  }
+  if (!isFallback && !id) return <Custom404 />;
 
-  const {
-    data: postById,
-    isLoading,
-    isFetched,
-  } = useQuery<PostData>(
-    ['post', id],
-    () => request(`${apiUrl}/graphql`, POST_BY_ID_QUERY, { id }),
-    { initialData: postData, enabled: !!id && tokenRefreshed },
-  );
+  const { post, isLoading } = usePostById({
+    id,
+    options: { initialData: postData },
+  });
```

Comment:
This might make a difference, compared to what I just did.
Perhaps you can check that out.